nomos-services/data-availability/sampling/src/network/adapters/libp2p.rs

```diff
@@ -73,19 +60,16 @@ where
         Self {
             settings,
             network_relay,
-            pending_sampling: HashMap::new(),
-            rng: ChaCha20Rng::from_entropy(),
         }
     }
 
-    async fn start_sampling(&mut self, blob_id: BlobId) -> Result<(), DynError> {
-        let mut ctx: SamplingContext = SamplingContext {
-            blob_id: (blob_id),
-            subnets: vec![],
-        };
-        for _i in 1..self.settings.num_samples {
-            let subnetwork_id = self.rng.gen_range(0..self.settings.subnet_size);
-            ctx.subnets.push(subnetwork_id);
+    async fn start_sampling(
+        &mut self,
+        blob_id: BlobId,
+        subnets: &Vec<SubnetworkId>,
```

Comment:
Nipick: is more rusty that if you are not gonna consume the `Vec` you can use a slice (Vec dereferences into slice). 
```rust
subnets: &[SubnetworkId]
```